### ISSUE TITLE
Error when deleting several unsaved files from project while open in editor

### ISSUE THREAD

**Issue by jrrhyne** [spyder-ide/spyder#20998]
### What steps will reproduce the problem?

Deleted a file from the project while the file was opened. 
After replying yes to the popup to close the file, this error occurred.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\jrhyne\anaconda3\envs\BACMTool\lib\site-packages\spyder\plugins\editor\plugin.py", line 2510, in removed
    self.close_file_from_name(filename)
  File "C:\Users\jrhyne\anaconda3\envs\BACMTool\lib\site-packages\spyder\plugins\editor\plugin.py", line 2506, in close_file_from_name
    self.editorstacks[0].close_file(index)
  File "C:\Users\jrhyne\anaconda3\envs\BACMTool\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1642, in close_file
    is_ok = (force or self.save_if_changed(cancelable=True, index=index)
  File "C:\Users\jrhyne\anaconda3\envs\BACMTool\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1819, in save_if_changed
    finfo = self.data[index]
IndexError: list index out of range
```

## Versions

* Spyder version: 5.4.3  (conda)
* Python version: 3.9.16 64-bit
* Qt version: 5.15.6
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                             :  1.4.1 (OK)
chardet >=2.0.0                                  :  5.1.0 (OK)
cloudpickle >=0.5.0                              :  2.2.1 (OK)
cookiecutter >=1.6.0                             :  2.1.1 (OK)
diff_match_patch >=20181111                      :  20200713 (OK)
intervaltree >=3.0.2                             :  3.0.2 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0 :  8.12.0 (OK)
jedi >=0.17.2,<0.19.0                            :  0.18.2 (OK)
jellyfish >=0.7                                  :  0.9.0 (OK)
jsonschema >=3.2.0                               :  4.17.3 (OK)
keyring >=17.0.0                                 :  23.13.1 (OK)
nbconvert >=4.0                                  :  7.3.1 (OK)
numpydoc >=0.6.0                                 :  1.5.0 (OK)
paramiko >=2.4.0                                 :  3.1.0 (OK)
parso >=0.7.0,<0.9.0                             :  0.8.3 (OK)
pexpect >=4.4.0                                  :  4.8.0 (OK)
pickleshare >=0.4                                :  0.7.5 (OK)
psutil >=5.3                                     :  5.9.5 (OK)
pygments >=2.0                                   :  2.15.1 (OK)
pylint >=2.5.0,<3.0                              :  2.17.2 (OK)
pylint_venv >=2.1.1                              :  3.0.1 (OK)
pyls_spyder >=0.4.0                              :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                             :  1.7.2 (OK)
pylsp_black >=1.2.0                              :  1.2.1 (OK)
qdarkstyle >=3.0.2,<3.2.0                        :  3.1 (OK)
qstylizer >=0.2.2                                :  0.2.2 (OK)
qtawesome >=1.2.1                                :  1.2.3 (OK)
qtconsole >=5.4.2,<5.5.0                         :  5.4.2 (OK)
qtpy >=2.1.0                                     :  2.3.1 (OK)
rtree >=0.9.7                                    :  1.0.1 (OK)
setuptools >=49.6.0                              :  67.6.1 (OK)
sphinx >=0.6.6                                   :  6.1.3 (OK)
spyder_kernels >=2.4.3,<2.5.0                    :  2.4.3 (OK)
textdistance >=4.2.0                             :  4.5.0 (OK)
three_merge >=0.1.1                              :  0.1.1 (OK)
watchdog >=0.10.3                                :  3.0.0 (OK)
zmq >=22.1.0                                     :  25.0.2 (OK)

# Optional:
cython >=0.21                                    :  None (NOK)
matplotlib >=3.0.0                               :  None (NOK)
numpy >=1.7                                      :  1.24.2 (OK)
pandas >=1.1.1                                   :  None (NOK)
scipy >=0.17.0                                   :  None (NOK)
sympy >=0.7.3                                    :  None (NOK)
```

**Comment by ccordoba12** [comment]
Hey @jrrhyne, thanks for reporting. I was able to reproduce your error on my side. We'll implement a fix for it in our next version (5.4.4), to be available in a couple of weeks.

**Comment by ccordoba12** [comment]
This was fixed by #21003.
